torch/custom_class.h

```diff
@@ -58,16 +61,21 @@ class class_ {
 
     // We currently represent custom classes as torchscript classes with a
     // capsule attribute
-    classTypePtr =
-        ClassType::create(c10::QualifiedName(qualClassName), classCU());
-    classTypePtr->addAttribute("capsule", CapsuleType::get());
-
-    c10::getCustomClassTypeMap().insert({typeid(c10::intrusive_ptr<CurClass>).name(),
-                              c10::StrongTypePtr(classCU(), classTypePtr)});
-    c10::getCustomClassTypeMap().insert({typeid(c10::tagged_capsule<CurClass>).name(),
-                              c10::StrongTypePtr(classCU(), classTypePtr)});
-
-    classCU()->register_type(classTypePtr);
+    classTypePtr = at::ClassType::create(
+        c10::QualifiedName(qualClassName),
+        std::weak_ptr<script::CompilationUnit>());
+    classTypePtr->addAttribute("capsule", at::CapsuleType::get());
+
+    c10::getCustomClassTypeMap().insert(
+        {typeid(c10::intrusive_ptr<CurClass>).name(),
```

Comment:
do you need demangle here and everywhere? Also - do we need to rely on RTTI for it? you could also use c10::get_fully_qualified_type_name (not sure where exactly we use these names). Or we never use those names other than for accounting? (in which case maybe we should just use typeid directly)